openshift-knative-operator/pkg/monitoring/monitoring_eventing_test.go

```diff
@@ -19,8 +19,15 @@ func TestLoadPlatformEventingMonitoringManifests(t *testing.T) {
 		t.Errorf("Got %d, want %d", len(manifests), 1)
 	}
 	resources := manifests[0].Resources()
-	if len(resources) != 28 {
-		t.Errorf("Got %d, want %d", len(resources), 28)
+
+	// We create a service monitor and a service monitor service per deployment.
+	// One clusterrolebinding (except for mt-broker-controller) per deployment for allowing tokenreviews, subjectaccessreviews
+	// to be used by kube proxy. All but one deployments have a different sa.
+	// Five RBAC resources from rbac-proxy.yaml
+	expectedEventingMonitoringResources := len(eventingDeployments)*2 + len(eventingDeployments) + 4
```

Comment:
what's the `4` from?

Reply:
2*len(eventingDeployments) + len(eventingDeployments) -1 + 5

Reply:
We have len(eventingDeployments) number of deployments and we create a clusterrolebinding per deployment except for one (`mt-broker-controller`). Thus len(eventingDeployments) -1. Then 5-1=4.

Reply:
mind adding a comment here, with your explanation ? 

I am sure a new person might not directly get the math behind this line

Reply:
I think it is trivial math but ok.

Reply:
I am sure folks missing context disagree that this calculation is trivial to undrstand

Reply:
The math might be trivial but as @matzew mentions the context is important. Also, I don't understand why you have 4 and not `len(eventingDeployments) - 1` as you have described. That kind of hardcodes again the numbed of eventing deployments.

Reply:
can we reflect this better or more dynamically, via code, rather than `*2`  and `+4` ?

Reply:
@rhuss the static resources part is irrelevant:
```
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: knative-prometheus-k8s
rules:
  - apiGroups:
      - ""
    resources:
      - services
      - endpoints
      - pods
    verbs:
      - get
      - list
      - watch
---
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  name: knative-prometheus-k8s
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: knative-prometheus-k8s
subjects:
  - kind: ServiceAccount
    name: prometheus-k8s
    namespace: openshift-monitoring
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: rbac-proxy-metrics-prom
rules:
  - nonResourceURLs: ["/metrics"]
    verbs: ["get"]
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: rbac-proxy-metrics-prom-rb
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: rbac-proxy-metrics-prom
subjects:
  - kind: ServiceAccount
    name: prometheus-k8s
    namespace: openshift-monitoring
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: rbac-proxy-reviews-prom
rules:
  - apiGroups: ["authentication.k8s.io"]
    resources:
      - tokenreviews
    verbs: ["create"]
  - apiGroups: ["authorization.k8s.io"]
    resources:
      - subjectaccessreviews
    verbs: ["create"]
```